index.js

```diff
@@ -108,6 +158,17 @@ class FSMerge {
     return [...new Set(result)];
   }
 
+  readdir(dirPath, callback) {
+    let result = [];
+    let promise = new Promise((res) => {
+      result = this.readdirSync(dirPath);
+      res(result);
+    });
+    promise.then((dirList) => {
```

Comment:
I'm not sure the API here is correct, I . believe the expected API in node for async is an errback `(error, value) =>`. Example: `readdir(dir, (err, dirList) => /*...*/)` currently you have `readdir(dir, (dirList) => /*...*/)`

Reply:
Thanks for pointing out. Fixed the issue. Added more strict tests.